### PR TITLE
oic: Make /oic/d and /oic/p resources discoverable

### PR DESCRIPTION
**Pull request by otaviobp**
In iotivity both resources are discoverable. Make them discoverable in
soletta to be compatible.

Signed-off-by: Otavio Pontes <otavio.pontes@intel.com>

### CODE REVIEW

File: src/lib/comms/sol-oic-server.c [hunk 1]
```diff
@@ -754,6 +681,21 @@ sol_oic_server_add_resource(const struct sol_oic_resource_type *rt,
     return NULL;
 }
 
+SOL_API struct sol_oic_server_resource *
+sol_oic_server_add_resource(const struct sol_oic_resource_type *rt,
+    const void *handler_data, enum sol_oic_resource_flag flags)
```
Comment (poussa) [inline src/lib/comms/sol-oic-server.c]: I would name this function to: sol_oic_server_resource_auto and the sol_oic_server_resource_full to sol_oic_server_resource.

Comment (otaviobp) [inline src/lib/comms/sol-oic-server.c]: good point.

Comment (gabrielschulhof) [inline src/lib/comms/sol-oic-server.c]: Alternatively, why not add the path to the ```struct sol_oic_resource_type``` and, if empty, just generate one. No need for two APIs then. Wouldn't even break ABI if the path is at the end of the structure.

Comment (poussa) [inline src/lib/comms/sol-oic-server.c]: Yes, that is better (IMO)

Comment (cabelitos) [inline src/lib/comms/sol-oic-server.c]: I Aggree

Comment (otaviobp) [inline src/lib/comms/sol-oic-server.c]: really better. I'll do it.

**Pull request closed without merging**